File: JavaCourceCommonApp/app/src/main/java/com/example/javacourcecommonapp/di/RepositoryModule.java
package com.example.javacourcecommonapp.di;

import com.example.javacourcecommonapp.data.api.MovieDataService;
import com.example.javacourcecommonapp.data.local.MovieDao;
import com.example.javacourcecommonapp.data.repository.MovieLocalRepository;
import com.example.javacourcecommonapp.data.repository.MovieLocalRepositoryImpl;
import com.example.javacourcecommonapp.data.repository.MovieRemoteRepository;
import com.example.javacourcecommonapp.data.repository.MovieRemoteRepositoryImpl;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class RepositoryModule {

    @Provides
    @Singleton
    MovieRemoteRepository provideMovieRepository(MovieDataService movieDataService){
        return new MovieRemoteRepositoryImpl(movieDataService);
    }

    @Provides
    @Singleton
    MovieLocalRepository provideMovieLocalRepository(MovieDao movieDao){
        return new MovieLocalRepositoryImpl(movieDao);
    }
}

File: Section 18 NewsAPIClient/app/src/main/java/com/anushka/newsapiclient/presentation/di/RepositoryModule.kt
package com.anushka.newsapiclient.presentation.di

import com.anushka.newsapiclient.data.repository.NewsRepositoryImpl
import com.anushka.newsapiclient.data.repository.dataSource.NewsRemoteDataSource
import com.anushka.newsapiclient.data.repository.dataSourceImpl.NewsRemoteDataSourceImpl
import com.anushka.newsapiclient.domain.repository.NewsRepository
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ApplicationComponent
import javax.inject.Singleton

@Module
@InstallIn(ApplicationComponent::class)
class RepositoryModule {

    @Singleton
    @Provides
    fun provideNewsRepository(
        newsRemoteDataSource: NewsRemoteDataSource
    ):NewsRepository{
        return NewsRepositoryImpl(newsRemoteDataSource)
    }

}














